mcs/tools/al/Al.cs

```diff
@@ -765,6 +729,83 @@ private void SetKeyPair (AssemblyName aname)
 			}
 		}
 
+		private void ReadCustomAttributesFromTemplateFile (string templateFile, ref AssemblyName aname)
+		{
+			// LAMESPEC: according to MSDN, the template assembly must have a
+			// strong name but this is not enforced
+			var options = IKR.UniverseOptions.MetadataOnly;
+
+			var universe = new IKR.Universe (options);
+			var asm = universe.LoadFile (templateFile);
+
+			// Create missing assemblies, we don't want to load them!
```

Comment:
It's better to use ResolveMissingMembers and then use __IsMissing for attribute type check